File: rb-appscript/tags/rb-appscript-0.5.3/test/test_appscriptreference.rb
#!/usr/bin/ruby -w

begin; require 'rubygems'; rescue LoadError; end

require 'test/unit'
require "appscript"

class TC_AppscriptReferences < Test::Unit::TestCase

	def setup
		@te = Appscript.app('TextEdit')
		@s = @te.to_s
	end

	def test_reference_forms
		[
			
			[@te.documents[
					Appscript.con.documents[3],
					Appscript.con.documents['foo']], 
					@s+'.documents[' +
					'con.documents[3], ' +
					'con.documents["foo"]]', nil],
					
					
					
			[@te.text, @s+'.text', nil],
			
			[@te.documents, @s+'.documents', nil],
			
			[@te.documents[1], 
					@s+'.documents[1]', nil],
			[@te.documents['foo'], 
					@s+'.documents["foo"]', nil],
			[@te.documents.ID(300), 
					@s+'.documents.ID(300)', nil],
			
			[@te.documents.next(:document), 
					@s+'.documents.next(:document)', nil],
			[@te.documents.previous(:document), 
					@s+'.documents.previous(:document)', nil],
			
			[@te.documents.first, @s+'.documents.first', nil],
			[@te.documents.middle, @s+'.documents.middle', nil],
			[@te.documents.last, @s+'.documents.last', nil],
			[@te.documents.any, @s+'.documents.any', nil],
			
			[Appscript.con.documents[3], 'con.documents[3]', nil],
			
			
			[Appscript.its.name.eq('foo').and(Appscript.its.words.eq([])), 
					'its.name.eq("foo").and(its.words.eq([]))', nil],
			
			[Appscript.its.words.ne([]), 
					'its.words.ne([])', 
					Appscript.its.words.eq([]).not], # i.e. there isn't a KAENotEqual operator, so not-equal tests are actually packed as an equal test followed by not test
					
			[Appscript.its.words.eq(nil), 'its.words.eq(nil)', nil],
			[Appscript.its.words.size.gt(0), 
					'its.words.size.gt(0)', nil],
			[Appscript.its.words.le(''), 'its.words.le("")', nil],
			[Appscript.its.words.begins_with('foo').not, 
					'its.words.begins_with("foo").not', nil],
			
			
			[Appscript.its.words.contains('foo'), 'its.words.contains("foo")', nil],
			[Appscript.its.words.is_in('foo'), 'its.words.is_in("foo")', nil],
			
			[@te.documents[Appscript.its.size.ge(42)], 
					@s+'.documents[its.size.ge(42)]', nil],
			
			[@te.documents[1, 'foo'], 
					@s+'.documents[1, "foo"]',
					@te.documents[Appscript.con.documents[1], Appscript.con.documents["foo"]]],
			
			[@te.documents[1].text \
					.paragraphs.characters[
							Appscript.con.characters[3], 
							Appscript.con.characters[55]
					].next(:character).after,
					@s+'.documents[1].text.paragraphs.characters' +
					'[con.characters[3], con.characters[55]]' +
					'.next(:character).after', nil],
			
			[Appscript.its.name.ne('foo').and(Appscript.its.words.eq([])).not,
					'its.name.ne("foo").and(its.words.eq([])).not',
					Appscript.its.name.eq('foo').not.and(Appscript.its.words.eq([])).not],
			
			[@te.documents.beginning, @s+'.documents.beginning', nil],
			[@te.documents.end, @s+'.documents.end', nil],
			[@te.documents[3].before, @s+'.documents[3].before', nil],
			[@te.documents['foo'].after, @s+'.documents["foo"].after', nil],
			
		].each do |val, res, unpacked_version|
			assert_equal(res, val.to_s)
			d = @te.AS_app_data.pack(val)
			val = unpacked_version ? unpacked_version : val
			val2 = @te.AS_app_data.unpack(d)
			if val.class == @te.AS_app_data.unpack(d).class # note: Appscript::Reference and Appscript::GenericReference currently aren't comparable with each other, so the next test would always fail for those
				assert_equal(val, val2)
				assert_block { val.eql?(val2) }
			end
		end
	end
end

File: py-appscript/tags/py-appscript-0.21.0/appscript_3x/test/test_appscriptreference.py
#!/usr/bin/env python3

import unittest
import appscript as AS


class TC_AppscriptReferences(unittest.TestCase):

	def setUp(self):
		self.te = AS.app('TextEdit')
		self.s = repr(self.te)

	def test_referenceForms(self):
		for val, res, unpackedVersion in [
				[self.te.text, self.s+'.text', None],
				
				[self.te.documents, self.s+'.documents', None],
				
				[self.te.documents[1], 
						self.s+'.documents[1]', None],
				[self.te.documents['foo'], 
						self.s+".documents['foo']", None],
				[self.te.documents.ID(300), 
						self.s+'.documents.ID(300)', None],
				
				[self.te.documents.next(AS.k.document), 
						self.s+'.documents.next(k.document)', None],
				[self.te.documents.previous(AS.k.document), 
						self.s+'.documents.previous(k.document)', None],
				
				[self.te.documents.first, self.s+'.documents.first', None],
				[self.te.documents.middle, self.s+'.documents.middle', None],
				[self.te.documents.last, self.s+'.documents.last', None],
				[self.te.documents.any, self.s+'.documents.any', None],
				
				[AS.con.documents[3], 'con.documents[3]', None],
				
				[self.te.documents[
						AS.con.documents[3]:
						AS.con.documents['foo']], 
						self.s+'.documents[' +
						'con.documents[3]:' +
						"con.documents['foo']]", None],
				
				
				[(AS.its.name == 'foo').AND(AS.its.words == []), 
						"(its.name == 'foo').AND(its.words == [])", None],
				
				[AS.its.words != [], 
						'its.words != []', 
						(AS.its.words == []).NOT], # i.e. there isn't a KAENotEqual operator, so not-equal tests are actually packed as an equal test followed by not test
						
				[AS.its.words == None, 'its.words == None', None],
				[AS.its.words.size > 0, 
						'its.words.size > 0', None],
				[AS.its.words <= '', "its.words <= ''", None],
				[AS.its.words.beginswith('foo').NOT, 
						"(its.words.beginswith('foo')).NOT", None],
				
				
				[AS.its.words.contains('foo'), "its.words.contains('foo')", None],
				[AS.its.words.isin('foo'), "its.words.isin('foo')", None],
				
				[self.te.documents[AS.its.size >= 42], 
						self.s+'.documents[its.size >= 42]', None],
				
				[self.te.documents[1:'bar'], 
						self.s+".documents[1:'bar']", 
						self.te.documents[AS.con.documents[1]:AS.con.documents['bar']]],
				
				[self.te.documents[1].text \
						.paragraphs.characters[
								AS.con.characters[3]:
								AS.con.characters[55]
						].next(AS.k.character).after,
						self.s+'.documents[1].text.paragraphs.characters' +
						'[con.characters[3]:con.characters[55]]' +
						'.next(k.character).after', None],
				
				[(AS.its.name != 'foo').AND(AS.its.words == []).NOT,
						"((its.name != 'foo').AND(its.words == [])).NOT",
						(AS.its.name == 'foo').NOT.AND(AS.its.words == []).NOT],
				
				[self.te.documents.beginning, self.s+'.documents.beginning', None],
				[self.te.documents.end, self.s+'.documents.end', None],
				[self.te.documents[3].before, self.s+'.documents[3].before', None],
				[self.te.documents['foo'].after, self.s+".documents['foo'].after", None],
				
				]:
			
			self.assertEqual(res, repr(val))
			d = self.te.AS_appdata.pack(val)
			val = unpackedVersion or val
			val2 = self.te.AS_appdata.unpack(d)
			if val.__class__ == self.te.AS_appdata.unpack(d).__class__: # note: Reference and GenericReference currently aren't comparable with each other, so the next test would always fail for those
				self.assertEqual(val, val2)
			val2 = self.te.AS_appdata.unpack(d)
			if val.__class__ == self.te.AS_appdata.unpack(d).__class__: # note: Reference and GenericReference currently aren't comparable with each other, so the next test would always fail for those
				self.assertEqual(val2, val)



if __name__ == '__main__':
	unittest.main()
